src/plugins.rb

```diff
@@ -83,6 +83,7 @@ def all_plugin_names
       end
 
       def plugin_directories
+        return [] unless plugins_dir.exist?
```

Comment:
It can. Not sure if this would be an improvement though? We basically have the normal code path and the special short circuit case for when a repo lacks the `plugins/` dir. Not sure if an alternative way of writing the same thing would be an improvement, and I don't think it matters much either way.

Reply:
Doesn't super matter, it just gives me the warm and fuzzies when ever I can do some complex ruby code in a single line.

Reply:
lol - have you considered taking up Perl? :grimacing: